addons/mail/models/mail_render_mixin.py

```diff
@@ -162,7 +162,7 @@ def _sub_relative2absolute(match):
         _sub_relative2absolute.base_url = base_url
         html = re.sub(r"""(<img(?=\s)[^>]*\ssrc=")(/[^/][^"]+)""", _sub_relative2absolute, html)
         html = re.sub(r"""(<a(?=\s)[^>]*\shref=")(/[^/][^"]+)""", _sub_relative2absolute, html)
-        html = re.sub(r"""(<[^>]+\bstyle="[^"]+\burl\('?)(/[^/'][^'")]+)""", _sub_relative2absolute, html)
+        html = re.sub(r"""(<[^>]+\bstyle="[^"]+\burl\((?:&\#34;|')?)(/(?:[^/']|(?!&\#34;))(?:[^'")]|(?!&\#34;))+)""", _sub_relative2absolute, html)
```

Comment:
This has gone from "difficult to read" to "line noise", the `re.VERBOSE` flag needs to be enabled and functional segments of the regexp split apart and commented.

Also reading 84562d3a1ad62e9bee9ca69e623034f4b7b0af89 why is the decimal escape version specifically supported and not the hexadecimal or textual ones? And what about dquotes?

Also is there really a good reason to split that from the introduction of the test? (cdb2c7656efe9d786dd4c8764120bef0665acf5e)

Reply:
I asked to split test commit from regex, to highlight what is supported in addition to what was already supported. Otherwise it is less clear what is new and what was already supported. Modifying the regex should simply bring a new set of URLs to test in the existing test.

Reply:
Deal with it :sunglasses: